### PR TITLE
Proposal for balancing node groups for multizone

### PR DESCRIPTION
**Pull request by MaciekPytel**
Ref: #47 

@mumoshu - does that look good to you?

### CODE REVIEW

File: cluster-autoscaler/proposals/balance_similar.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+Pros:
+ * Simplest solution.
+
+Cons:
+ * If at least a single pending pod uses zone-based scheduling features the
+   whole scale-up will likely go to a single node group.
+ * We add slightly different nodes than those chosen by expansion.Strategy. In
+   particular any zone-based choices made by expansion.Strategy will be
+   discarded.
+ * We operate on single node groups when creating expansion options, so maximum
```
Comment (mwielgus) [inline cluster-autoscaler/proposals/balance_similar.md]: We could do split before capping the size.

Comment (MaciekPytel) [inline cluster-autoscaler/proposals/balance_similar.md]: The problem is that if we do the capping after using expander.Strategy, than the strategy cannot use the number of nodes as input. That seems like a massive limitation.

Alternatively we could do all the balancing before applying strategy, which is exactly what [S2] is. However, having a scale up that won't fit in a single node group feels like an edge case and I think it's not worth all the extra complexity. Especially given that we will just do another scale up in 10s.

File: cluster-autoscaler/proposals/balance_similar.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+   correct node group.
+ * Preferably we should avoid implementing this using existing
+   expansion.Strategy interface, as that would likely require non-backward
+   compatible refactor of the interface and we want to allow using this feature
+   with different strategies.
+ * User must be able to disable this logic using a flag.
+
+## General idea
+The general idea behind this proposal is to introduce a concept of "Node Group
+Set", consisting of one or more of "similar" node groups (definition of
```
Comment (MaciekPytel) [inline cluster-autoscaler/proposals/balance_similar.md]: done

**Pull request merged**